ocs_ci/ocs/resources/pvc.py

```diff
@@ -122,17 +124,37 @@ def reclaim_policy(self):
             'persistentVolumeReclaimPolicy'
         )
 
+    @retry(UnexpectedBehaviour, tries=5, delay=2, backoff=1)
     def resize_pvc(self, new_size, verify=False):
         """
-        Returns the PVC size pvc_name in namespace
+        Modify the capacity of PVC
+
+        Args:
+            new_size (int): New size of PVC in Gi
+            verify (bool): True to verify the change is reflected on PVC,
+                False otherwise
 
         Returns:
             bool: True if operation succeeded, False otherwise
         """
-        self.data['status']['capacity']['storage'] = f"{new_size}Gi"
-        self.apply(**self.data)
+        patch_param = f'{{"spec": {{"resources": {{"requests": {{"storage": "{new_size}Gi"}}}}}}}}'
+
+        # Modify size of PVC
+        assert self.ocp.patch(resource_name=self.name, params=patch_param), (
+            f"Patch command to modify size of PVC {self.name} has failed."
+        )
+
         if verify:
-            return self.size == new_size
+            self.reload()
+            if self.size != new_size:
+                raise UnexpectedBehaviour(
```

Comment:
if this is the reason for the retry (meaning to "wait" for the action to take place) so I guess you should use timeoutingSampler because retry will call 2nd resize which is not the intention

Reply:
ah.. I see. will change it to timeoutingSampler.